astronomer/providers/snowflake/hooks/snowflake_sql_api.py

```diff
@@ -0,0 +1,241 @@
+import uuid
+from abc import ABC
+from datetime import timedelta
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import aiohttp
+import requests
+from airflow import AirflowException
+from airflow.providers.snowflake.hooks.snowflake import SnowflakeHook
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization
+
+from astronomer.providers.snowflake.hooks.sql_api_generate_jwt import JWTGenerator
+
+
+class SnowflakeSqlApiHookAsync(SnowflakeHook, ABC):
+    """
+    A client to interact with Snowflake using SQL API  and allows submitting
+    multiple SQL statements in a single request. In combination with aiohttp, make post request to submit SQL
+    statements for execution, poll to check the status of the execution of a statement. Fetch query results
+    asynchronously.
+
+    This hook requires the snowflake_conn_id connection. This hooks mainly uses account, schema, database, warehouse,
+    private_key_file or private_key_content field must be setup in the connection.
+    Other inputs can be defined in the connection or hook instantiation.
+
+    :param snowflake_conn_id: Reference to
+        :ref:`Snowflake connection id<howto/connection:snowflake>`
+    :param account: snowflake account name
+    :param authenticator: authenticator for Snowflake.
+        'snowflake' (default) to use the internal Snowflake authenticator
+        'externalbrowser' to authenticate using your web browser and
+        Okta, ADFS or any other SAML 2.0-compliant identify provider
+        (IdP) that has been defined for your account
+        'https://<your_okta_account_name>.okta.com' to authenticate
+        through native Okta.
+    :param warehouse: name of snowflake warehouse
+    :param database: name of snowflake database
+    :param region: name of snowflake region
+    :param role: name of snowflake role
+    :param schema: name of snowflake schema
+    :param session_parameters: You can set session-level parameters at
+        the time you connect to Snowflake
+    :param token_life_time: lifetime of the JWT Token
+    :param token_renewal_delta: Renewal time of the JWT Token
+    """
+
+    LIFETIME = timedelta(minutes=59)  # The tokens will have a 59 minute lifetime
+    RENEWAL_DELTA = timedelta(minutes=54)  # Tokens will be renewed after 54 minutes
+
+    def __init__(
+        self,
+        snowflake_conn_id: str,
+        token_life_time: timedelta = LIFETIME,
+        token_renewal_delta: timedelta = RENEWAL_DELTA,
+        *args: Any,
+        **kwargs: Any,
+    ):
+        self.snowflake_conn_id = snowflake_conn_id
+        self.token_life_time = token_life_time
+        self.token_renewal_delta = token_renewal_delta
+        super().__init__(snowflake_conn_id, *args, **kwargs)
+        self.private_key = None
+
+    def get_private_key(self) -> None:
```

Comment:
can't we just call `_get_conn_params` of SnowflakeHook and then get the private key
```
 conn_config = self._get_conn_params()
return conn_config['private_key'] 
```

Reply:
No SnowflakeHook `_get_conn_params` tries different way of decoding the private key